model/event/event_finder.go

```diff
@@ -61,6 +63,36 @@ func CountProcessEvents(taskID string) (int, error) {
 	return db.CountQ(TaskLogCollection, db.Query(filter))
 }
 
+func FindLastProcessedEvent() (*EventLogEntry, error) {
+	q := db.Query(bson.M{
+		processedAtKey: bson.M{
+			"$exists": true,
+		},
+	}).Sort([]string{"-" + processedAtKey})
+
+	e := EventLogEntry{}
+	if err := db.FindOneQ(AllLogCollection, q, &e); err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "failed to fetch most recently processed event")
+	}
+
+	return &e, nil
+}
+
+func CountUnprocessedEvents() (int, error) {
+	q := db.Query(UnprocessedEvents())
```

Comment:
does unprocessed events get used by anything outside of the events package?

Reply:
Not in this commit, but in #972 it's used.

Reply:
probably should still do this.